lib/dal/src/schema/variant/definition.rs

```diff
@@ -762,3 +764,54 @@ pub struct SiPropValueFrom {
 //         }
 //     }
 // }
+
+#[derive(Deserialize, Serialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct SchemaVariantDefinitionView {
+    id: SchemaVariantId,
+    name: String,
+    schema_variant_id: SchemaVariantId,
+    category: String,
+    #[serde(alias = "menu_name")]
+    menu_name: Option<String>,
+    color: String,
+    component_type: ComponentType,
+    link: Option<String>,
+    description: Option<String>,
+    #[serde(flatten)]
+    timestamp: Timestamp,
+}
+
+impl SchemaVariantDefinitionView {
+    pub async fn list(ctx: &DalContext) -> SchemaVariantDefinitionResult<Vec<Self>> {
+        let mut views = Vec::new();
+
+        let schemas = Schema::list(ctx).await?;
+        for schema in schemas {
+            let default_schema_variant =
+                SchemaVariant::get_default_for_schema(ctx, schema.id()).await?;
+            views.push(SchemaVariantDefinitionView {
+                id: default_schema_variant.id,
+                name: schema.name.to_owned(),
+                schema_variant_id: default_schema_variant.id.to_owned(),
+                category: default_schema_variant.category.to_owned(),
+                color: default_schema_variant
+                    .get_color(ctx)
+                    .await?
+                    .unwrap_or("#0F0F0F".into()),
+                timestamp: default_schema_variant.timestamp.to_owned(),
+                component_type: default_schema_variant
+                    .get_type(ctx)
+                    .await?
+                    .unwrap_or(ComponentType::Component),
+                link: default_schema_variant.link.to_owned(),
+                // TODO: Schema Variants need a description
+                description: None,
+                // TODO: What is the difference between a category and a menu name?
+                menu_name: None,
```

Comment:
I'm guessing nothing, but we'll clean this up after I get through `NodeAddMenu` removal.